testing/file_test/file_test_base.cpp

```diff
@@ -45,6 +65,38 @@ ABSL_FLAG(bool, dump_output, false,
 
 namespace Carbon::Testing {
 
+// Information for a test case.
+struct FileTestInfo {
+  // The name.
+  std::string test_name;
+
+  // A factory function for creating the test object.
+  std::function<auto()->FileTestBase*> factory_fn;
+
+  // gtest's information about the test.
+  ::testing::TestInfo* registered_test;
+
+  // The test result, set after running.
+  std::optional<ErrorOr<TestFile>> test_result;
+
+  // Whether autoupdate would change (or when autoupdating, already changed)
+  // test expectations.
+  bool autoupdate_differs = false;
+};
+
+// Adapts a `FileTestBase` instance to gtest for outputting results.
+class FileTestCase : public ::testing::Test {
```

Comment:
I notice you used a leading `::` here, but not in cases like `testing::ScopedTrace` below. Is there a reason for the difference?